### PR TITLE
code improvement on gameclient.cpp

### PR DESCRIPTION
**Pull request by Ryozuki**

### CODE REVIEW

File: src/game/client/gameclient.cpp [hunk 1]
```diff
@@ -2183,7 +2169,7 @@ void CLocalProjectile::Tick(int CurrentTick, int GameTickSpeed, int LocalClientI
 
 	bool OwnerCanProbablyHitOthers = (m_pWorld->m_Tuning[g_Config.m_ClDummy].m_PlayerCollision || m_pWorld->m_Tuning[g_Config.m_ClDummy].m_PlayerHooking);
 
-	if(((Target >= 0 && (m_Owner >= 0 ? OwnerCanProbablyHitOthers : 1 || Target == m_Owner)) || Collide || GameLayerClipped(CurPos)) && !IsWeaponCollide)
```
Comment (def-) [inline src/game/client/gameclient.cpp]: Are you trying to hide some secret change in here? ;)

Comment (Ryozuki) [reply src/game/client/gameclient.cpp]: it always evaluates to true :o

File: src/game/client/gameclient.cpp [hunk 2]
```diff
@@ -1518,125 +1512,117 @@ void CGameClient::OnPredict()
...
+                NewPresses = true;
+            }
+            if(FullAuto && (Input.m_Fire & 1))
+                WillFire = true;
+            if(!WillFire)
+                break;
+            if(!IsRace(&Info) && !m_Snap.m_pLocalCharacter->m_AmmoCount && Local->m_ActiveWeapon != WEAPON_HAMMER)
+                break;
+            int ExpectedStartTick = Tick - 1;
+            ReloadTimer = g_pData->m_Weapons.m_aId[Local->m_ActiveWeapon].m_Firedelay * SERVER_TICK_SPEED / 1000;
```
Comment (Learath2) [inline src/game/client/gameclient.cpp]: Some empty lines around here to break up the huge chunks.

**Comment by heinrich5991** [comment]
What is the intention behind making the casts to `int`/`bool`/`char` explicit? Do they generate warnings somewhere? Is there a different reason to make them explicit?

**Comment by def-** [comment]
We should enable clang-tidy checks if you want to keep stuff like this fixed.

File: src/game/client/gameclient.cpp [hunk 3]
```diff
@@ -1518,125 +1512,117 @@ void CGameClient::OnPredict()
...
-					ReloadTimer = g_pData->m_Weapons.m_aId[Local->m_ActiveWeapon].m_Firedelay * SERVER_TICK_SPEED / 1000;
-					if(!DirectInput)
-						ReloadTimer++;
-				}
-			} while(false);
-
-			if(ReloadTimer)
+			// handle weapons
+            if(ReloadTimer)
+                break;
```
[the remark this answers is not in the document]
Comment (Ryozuki) [reply src/game/client/gameclient.cpp]: it's actually inside another for loop

[the remark this answers is not in the document]
Comment (Ryozuki) [reply src/game/client/gameclient.cpp]: hmm true it shouldnt be like this

File: src/game/client/gameclient.cpp [hunk 4]
```diff
@@ -345,9 +345,9 @@ void CGameClient::OnInit()
 		for(unsigned int i = 0; i < 16; i++)
 		{
 			if(rand() % 2)
-				g_Config.m_ClTimeoutCode[i] = (rand() % 26) + 97;
+				g_Config.m_ClTimeoutCode[i] = static_cast<char>((rand() % 26) + 97);
```
[the remark this answers is not in the document]
Comment (heinrich5991) [reply src/game/client/gameclient.cpp]: We can move to C++-style casts if we want to, even gradually, but that should be an active decision, and not a PR that just changes some of the casts in the code base.

That you perceive the source code as inconsistent is not a valid reason for making it worse IMO.

[the remark this answers is not in the document]
Comment (Ryozuki) [reply src/game/client/gameclient.cpp]: You always make so many assumptions

File: src/game/client/gameclient.cpp [hunk 5]
```diff
@@ -1514,113 +1514,112 @@ void CGameClient::OnPredict()
...
+					if(!ReloadTimer)
+					{
+						ReloadTimer = g_pData->m_Weapons.m_aId[Local->m_ActiveWeapon].m_Firedelay * SERVER_TICK_SPEED / 1000;
+						if(!DirectInput)
+							ReloadTimer++;
+					}
+				}
+			}
 
             if(ReloadTimer)
```
Comment (Learath2) [inline src/game/client/gameclient.cpp]: Indent :)

File: src/game/client/gameclient.cpp [hunk 6]
```diff
@@ -1514,113 +1514,112 @@ void CGameClient::OnPredict()
...
+					if(!DirectInput)
+					{
+						ReloadTimer++;
+						ExpectedStartTick++;
+					}
+					switch(Local->m_ActiveWeapon)
+					{
+						case WEAPON_RIFLE:
+						case WEAPON_SHOTGUN:
+						case WEAPON_GUN:
```
Comment (Learath2) [inline src/game/client/gameclient.cpp]: Indentation for the case blocks are different. This one I don't really know what to do with to be honest. Don't think we have a rule for these anyway. @heinrich5991 do we?

Comment (heinrich5991) [reply src/game/client/gameclient.cpp]: I don't think we do. I think I've seen both indentations on switch level and one level deeper.

Comment (heinrich5991) [reply src/game/client/gameclient.cpp]: But I think we do always indent one level more for blocks inside case labels, so maybe this should go one to the left.

**Comment by heinrich5991** [comment]
I think these are good changes, :) except maybe for the explicit casts to char, bool and int.

**Comment by Learath2** [comment]
> I think these are good changes, :) except maybe for the explicit casts to char, bool and int.

What do you think these could cause? I'm for not touching these things and avoiding all redundant casts but these shouldn't change anything in behaviour.

**Pull request merged**